internal/tui/tui.go

```diff
@@ -280,6 +291,74 @@ func (a appModel) Update(msg tea.Msg) (tea.Model, tea.Cmd) {
 		}
 		return a, nil
 
+	case dialog.CloseSetupDialogMsg:
+		a.showSetupDialog = false
+
+		err := config.Update(func(cfg *config.Config) {
+			// Add Agent
+			if cfg.Agents == nil {
+				cfg.Agents = make(map[config.AgentName]config.Agent)
+			}
+			cfg.Agents[config.AgentPrimary] = config.Agent{
+				Model:     msg.Model.ID,
+				MaxTokens: msg.Model.DefaultMaxTokens,
+			}
+			cfg.Agents[config.AgentTitle] = config.Agent{
+				Model:     msg.Model.ID,
+				MaxTokens: 80,
+			}
+			cfg.Agents[config.AgentTask] = config.Agent{
+				Model:     msg.Model.ID,
+				MaxTokens: msg.Model.DefaultMaxTokens,
+			}
+
+			// Add Provider
+			if cfg.Providers == nil {
+				cfg.Providers = make(map[models.ModelProvider]config.Provider)
+			}
+
+			cfg.Providers[msg.Provider] = config.Provider{
+				APIKey: msg.APIKey,
+			}
+
+			cfg.SetupComplete = true
+		})
+		if err != nil {
+			slog.Debug("Failed to update config", "error", err)
+			panic(err)
+		}
+
+		// Reinitialize the agent
+		a.app.PrimaryAgent, err = agent.NewAgent(
+			config.AgentPrimary,
+			a.app.Sessions,
+			a.app.Messages,
+			agent.PrimaryAgentTools(
+				a.app.Permissions,
+				a.app.Sessions,
+				a.app.Messages,
+				a.app.History,
+				a.app.LSPClients,
+			),
+		)
+		if err != nil {
+			slog.Debug("Failed to initialize agent", "error", err)
+			panic(err)
+		}
```

Comment:
maybe we move this stuff somewhere else, out of the tui loop; when we have an API (non-TUI use case), we might even want to expose project init as a route, for instance. having the code live elsewhere seems good.